lib/openhab/core/emulate_hash.rb

```diff
@@ -0,0 +1,233 @@
+# frozen_string_literal: true
+
+require "forwardable"
+
+module OpenHAB
+  module Core
+    #
+    # Make an openHAB class acts like Ruby Hash.
+    #
+    module EmulateHash
+      include Enumerable
+      extend Forwardable
+
+      # The implementing class must implement #to_hash
+      def_delegators :to_hash,
+                     :any?,
+                     :compact,
+                     :compare_by_identity?,
+                     :deconstruct_keys,
+                     :each,
+                     :each_key,
+                     :each_pair,
+                     :each_value,
+                     :equals,
+                     :empty?,
+                     :filter,
+                     :flatten,
+                     :has_value?,
+                     :invert,
+                     :key,
+                     :keys,
+                     :length,
+                     :merge,
+                     :rassoc,
+                     :reject,
+                     :select,
+                     :shift,
+                     :size,
+                     :to_a,
+                     :to_h,
+                     :transform_keys,
+                     :transform_values,
+                     :values,
+                     :value?
+
+      # @!visibility private
+      def replace
+        raise NotImplementedError
+      end
+
+      # @!visibility private
+      def to_hash
+        raise NotImplementedError
+      end
+
+      # @!visibility private
+      def fetch(key)
+        raise NotImplementedError
```

Comment:
this can be delegated to `to_map` (except for stringifying the key). make sure to pass `*default, &block` through too though